### ISSUE TITLE
Domain Picker: Improve semantics

### ISSUE THREAD

**Issue by StefanNieuwenhuis** [Automattic/wp-calypso#51175]
#### A11y
Improving semantics will benefit a11y enormously since it's easier for both humans and machines to read and understand the HMTL-code. It'll also ensures equal access for people with disabilities and enables people to customize their experiences.  

#### Maintenance
Using semantic markup results in very clear, standardized code, and that means you’ll save time in the long run. 
Clear code is: 
* More easily maintained.
* Easier to work on without confusion (since the code is organized in a logical way).
* The appearance of the content is more easily changed and updated.
* Separated from its content, hence easier to update styles or add multiple types of data.

### What
With the `<DomainPicker/>` component, we face the following two semantic issues:

#### 1. The WrappingComponent returns a div when the _type prop_ is "button".

We have two variants of the Domain Picker's Suggestion Items.

**Variant 1: A list of buttons**
<img width="569" alt="Screenshot 2021-03-18 at 10 20 43" src="https://user-images.githubusercontent.com/12104589/111602633-d6693380-87d3-11eb-945f-74523b0b1e01.png">

**Variant 2: A list of divs**
<img width="546" alt="Screenshot 2021-03-18 at 10 21 33" src="https://user-images.githubusercontent.com/12104589/111602666-e123c880-87d3-11eb-977c-09e8041ec3fd.png">

The confusing part is that the second variant passes a "button" to the type prop of the `<WrappingComponent/>`. This is done to prevent the nesting of buttons in buttons (i.e. the "Select Button"), which is forbidden by the [w3c spec](https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element). See also https://github.com/Automattic/wp-calypso/pull/49249#discussion_r566798781

https://github.com/Automattic/wp-calypso/blob/a2c88dd7ae3b9dd04095b9132ebc790bf213e71c/packages/domain-picker/src/domain-picker/suggestion-item-wrapper.tsx#L23-L25

In order to make the Domain Picker better maintainable, we have to rethink the `<WrappingComponent/>`'s logic. 

#### 2. The InfoTooltip component renders a button inside a button
Like we've seen in the previous paragraph, the nesting of a button inside a button is forbidden by the [w3c spec](https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element), but this is exactly what happens when an InfoTooltip component is shown in SuggestionItems where the `itemType !== button`. 

When the `hsts_required` prop for a DomainSuggestion is `true`, a InfoTooltip that renders a [ButtonWithIcon ](https://github.com/Automattic/wp-calypso/blob/a2c88dd7ae3b9dd04095b9132ebc790bf213e71c/packages/domain-picker/src/info-tooltip/index.tsx#L50-L69) is injected to the DOM.
Combined with **variant 1** (see above), this results in a button nested in a button, i.e. invalid HTML. 

### A/C
* [ ] Tests (either unit or integration) should cover all "item types" (i.e. `button`, `row` & `radio`).

*Note that https://github.com/Automattic/wp-calypso/issues/47741 is blocked by this issue

**Comment by ciampo** [comment]
As discussed, 

> 2. The InfoTooltip component renders a button inside a button

should not be flagged as an issue.

One potential solution is to change the suggestion item's wrapping component to be a `<label>` element and be paired with a hidden `<input type="radio" />` element ([A11yPen](https://a11ypen.net/s/WEorwZArOk4Fpj2T0OpE))

We should have a conversation with the design team and understand if keeping 2 variations for the suggestion items is still necessary — as that could simplify our code substantially.
